lsp-mode.el

```diff
@@ -4482,6 +4546,11 @@ Added to `after-change-functions'."
   :group 'lsp-mode)
 
 
+(defvar lsp--on-change-fns-timer nil)
```

Comment:
What do we need that for?

Reply:
This is a timer to debounce `lsp-on-change-functions`. So we will only notify those on change information when really in idle and after idle delay.

Reply:
We should not debounce for this reason - we should try to apply the additional text edits as soon as possible. The longer delay the be bigger the chance to fail is. Also, it will be hard to reason about why the text edits are not applied immediately.

Reply:
If we eagerly apply the additional text edits, the document between client and server will be mismatched. That means the range returned by completion may completely un-applicable.
Thus I debounce this and apply it later on. At least make sure that while we're in middle of doing completion, the selected item can still be applied to correct range.